### PR TITLE
feat: Support matrix*, pmatrix*, bmatrix*, Bmatrix*, vmatrix*, and  Vmatrix*.

### PR DESCRIPTION
**Pull request by ronkok**
Addresses #2484.

### CODE REVIEW

**Comment by ronkok** [comment]
Screenshotter was throwing errors, so I deleted the screenshotter test for this PR. Now Screenshotter is still throwing errors even though this PR has no new screenshots. I'm baffled.

**Comment by ronkok** [comment]
Okay, I fixed the alignment bug and updated screenshots. This is now ready for reveiw.

**Review by kevinbarabash — APPROVED** [review]
LGTM.  Thanks for adding support for these `*` variants.  I think the code could be deduped a bit, but it isn't blocking.

File: src/environments/array.js [hunk 1]
```diff
@@ -788,6 +790,63 @@ defineEnvironment({
...
+        );
+        return delimiters ? {
+            type: "leftright",
+            mode: context.mode,
+            body: [res],
+            left: delimiters[0],
+            right: delimiters[1],
+            rightColor: undefined, // \right uninfluenced by \color in array
+        } : res;
+    },
```
[the remark this answers is not in the document]
Comment (ronkok) [reply src/environments/array.js]: I took a stab at a dedupe. It's not pretty code, but it is fewer lines than before.

File: src/environments/array.js [hunk 2]
```diff
@@ -788,6 +790,63 @@ defineEnvironment({
...
+            parser.consumeSpaces();
+            colAlign = parser.fetch().text;
+            if ("lcr".indexOf(colAlign) === -1) {
+                throw new ParseError("Expected l or c or r", parser.nextToken);
+            }
+            parser.consume();
+            parser.consumeSpaces();
+            if (parser.fetch().text !== "]") {
+                throw new ParseError("Expected ]", parser.nextToken);
+            }
```
[the remark this answers is not in the document]
Comment (ronkok) [reply src/environments/array.js]: Done.

**Comment by ronkok** [comment]
The dedupe is done and all tests are once again passing. @kevinbarabash perhaps you would like to cast an eye over the code once again.

**Comment by kevinbarabash** [comment]
@ronkok I had another look and it LGTM.  🙂

**Comment by cirosantilli** [comment]
:heart:

**Pull request merged**